README.md

```diff
@@ -1,17 +1,73 @@
+* [Environment branches with GitLab flow](http://docs.gitlab.com/ee/workflow/gitlab_flow.html#environment-branches-with-gitlab-flow)
 
-![develop jobs](images/dev-jobs.png)
+# [Production branch with GitLab flow](http://docs.gitlab.com/ee/workflow/gitlab_flow.html#production-branch-with-gitlab-flow)
+[Production branch with Gitlab flow](http://docs.gitlab.com/ee/workflow/gitlab_flow.html#production-branch-with-gitlab-flow) assumes two major branches.
 
-## staging(release candidates) jobs
-jobs for rc (rc stands for release candidates) branch.
+* `main` branch:  This branch is for tracking production 'grade' code, it can be seen as release candidates.  Not as github flow, master branch is not for tracking production codes.
+  * features branch: as standard git flow and github flow, this is for developing each feature and will be merged to master.
+* `production` branch:  This is the branch for tracking production code. Every commit on the branch will be shipped to the production environment.
 
-![develop jobs](images/staging-jobs.png)
+## [Concourse pipeline to support "Production branch with Gitlab flow"](ci/2-envs.yaml)
+![2-envs.yml](images/2-envs.png)
 
-## production jobs
-jobs for master branch.
+[2-envs.yml](ci/2-envs.yml) supports that:
+* Semantic versioning by [semver-resource](https://github.com/concourse/semver-resource)
+* Continuous Integration(CI) for pull requests.
+* Continuous Deployment(CD) of `master` branch
+  * deployment to `staging` environment bumps release candidate version (stored in `rc-version` resource)
+    * release candidate version will be of the form `x.y.z-rc.n`
+* CI/CD of `production` branch
+  * `merge-master-to-production` job is the trigger to start deployment to production.
+  * Succeeding jobs performs CI/CD
+   * `version-production` job always computes final version from `rc-version` resource and stores it to `final-version` resource.
+    * If release candidate version was `x.y.z-rc.n`, final version will be `x.y.z`
+   * `start-next-rc` is executed after `tag-production`.  This stores next release candidate version to `rc-version` resource.
+     * If `x.y.z` will be tagged, `start-next-rc` bumped patch version.  i.e. in this case, next release candidate version will be `x.y.(z+1)-dev.1`
+     * Team can manually bump minor/major version when big changes happened on master branch.
+* tagging commits and actual shipment jobs are separated.  This enables you to ship the latest commit on production branch to production environment __without__ bumping versions.  This would be useful for deployment failure or network issues.
 
-![develop jobs](images/production-jobs.png)
+Please note that this pipeline does NOT support hotfix-ing on master branch.  This means you need to commits those fixes to master branch and start over release process.  See [2-envs.yml](ci/2-envs.yml) for implementation details.
```

Comment:
I guess you'd like `hotfix-ing on production branch` instead of `hotfix-ing on master branch` here?